### DISCUSSION TITLE
Is it possible to add PAGE NUMBER to crossreference when render quarto to Word ?

### DISCUSSION THREAD

**Discussion by FannyVille** [quarto-dev/quarto-cli#6254] [Q&A]
### Description

Hi,

# My issue
In a rendered **word** document (not pdf),
I would like to know if there is a way to add after the label of the cross-reference, the page number.
Or at least just instead of giving the section name as label, give the page number. 

## Actual situation

```
## 2. Method {#sec-method}

We selected the method TRUCBIDULE ... 


## 3. Results {#sec-resultat}

  ### 3.1 Results part 1 {#sec-resultP1}

  ### 3.1 Results part 2 {#sec-resultP2}

Blabla bla ,  due to the selected method TRUCBIDULE  @sec-method p. 
```

The resulted Word document:
![image](https://github.com/quarto-dev/quarto-cli/assets/76000099/c77db39f-fe86-4d25-a11d-c34c5aaa9687)

## The desired solution
![image](https://github.com/quarto-dev/quarto-cli/assets/76000099/b809fe97-0678-471a-935c-e622b8c54cfd)

Thank you in advance

**Comment by mcanouil** [comment]
This would require to know in advance how many pages are going to be needed to write everything and/or to use very specific Word feature.
So I would say, based on the level of support from Pandoc, that it is not possible to do that right now.

Watch the following issue:

- https://github.com/jgm/pandoc/issues/8269

**Comment by FannyVille** [comment]
Hi

Thank you @mcanouil for replying very fast to questions. 

I found a solution !
I leave it here, if anyone else has the same problem as me

# My solution

Using the package `officer` and `officedown` and the function `officer::run_word_field` 

There is nothing to add in the YALM

1.  At the beginning of the script, below the YALM I am adding  a block code to load the `officer`  :

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
library(officedown)
library(officer)
```

2.  Then in my text after adding my cross-reference `@sec-method` I continue with inline code  _(p._`r run_word_field(field = "PAGEREF sec-method")` _)_


```
## 2. Method {#sec-method}

We selected the method TRUCBIDULE ... 


## 3. Results {#sec-resultat}

  ### 3.1 Results part 1 {#sec-resultP1}

  ### 3.1 Results part 2 {#sec-resultP2}

Blabla bla ,  due to the selected method TRUCBIDULE  @sec-method (p.`r run_word_field(field = "PAGEREF sec-method")`)

```

It is a bit boring/ long to write but it does the job. 

here the result: 

![image](https://github.com/quarto-dev/quarto-cli/assets/76000099/2481623f-afeb-468f-ab11-c06ce5ef774c)

# Improvement 

To be faster in my writing, I had the idea to creating a code snippets:
```
snippet p
	(p.` r run_word_field(field= "PAGEREF ${0:mysection}") `)

```
The problem is that there are subtleties I don't know about with snippets, but if I don't add spaces before and after the first backtick the snippets don't appear. So I had to add spaces. ``(p. ` r ``.  For inline code to work, there must be no gap between the first backtick and the `r` character. So every time I use my trick, I'll have to remember to remove the space to make the code work online.


## EDIT the issue with code snippets 

The code snippet below allow to avoid space after the `r` character by adding `${1:}r`
```
snippet p
	 (p.`${1:}r run_word_field(field= "PAGEREF ${2:mysection}")` 
```

**Reply by mcanouil** [reply]
The "Officeverse" is a good bet as it is focused on Word specific features as long as you want to work with `knitr`/R said the computing engine.
